yaml_database/src/yaml_schedule_database.cpp

```diff
@@ -0,0 +1,61 @@
+#include <yaml_database/yaml_schedule_database.h>
+
+#include <iostream>
+#include <string>
+
+namespace tabetai2::yaml_database {
+
+using namespace core::database;
+using namespace core::recipe;
+using namespace core::schedule;
+
+YamlScheduleDatabase::YamlScheduleDatabase(std::string database_file,
+                                           std::string database_name,
+                                           std::shared_ptr<RecipeRepository> recipe_repository)
+: YamlDatabase<Schedule>{std::move(database_file), std::move(database_name)},
+  m_recipe_repository{std::move(recipe_repository)} {
+
+}
+
+Schedule YamlScheduleDatabase::from_yaml(YAML::Node entry) const {
+    Schedule schedule{entry["id"].as<Id>(), entry["start_date"].as<std::string>()};
+    for (const auto& day_entry : entry["days"]) {
+        ScheduleDay day;
+        for (const auto& meal_entry : day_entry) {
+            auto recipe = m_recipe_repository->find_by_id(meal_entry["recipe"].as<Id>());
+            if (!recipe) {
+                std::cout << "Non-existing recipe referenced in schedule." << std::endl;
```

Comment:
Did you mean to throw or something after this line? Or just letting `recipe.value()` throw a `std::bad_optional_access`?